AVInput/AVInput.cpp

```diff
@@ -224,5 +313,913 @@ string AVInput::currentVideoMode(bool &success)
+}
+
+uint32_t AVInput::getInputDevicesWrapper(const JsonObject& parameters, JsonObject& response)
+{
+    LOGINFOMETHOD();
+
+    if (parameters.HasLabel("typeOfInput")) {
+        string sType = parameters["typeOfInput"].String();
+        int iType = 0;
+        try {
+            iType = getTypeOfInput (sType);
+        }catch (...) {
+            LOGWARN("Invalid Arguments");
+            return Core::ERROR_BAD_REQUEST;
+        }
+        response["devices"] = getInputDevices(iType);
+    }
+    else {
+        JsonArray listHdmi = getInputDevices(HDMI);
+        JsonArray listComposite = getInputDevices(COMPOSITE);
+        for (int i = 0; i < listComposite.Length(); i++) {
+            listHdmi.Add(listComposite.Get(i));
+        }
+        response["devices"] = listHdmi;
+    }
+    return Core::ERROR_NONE;
+}
+
+uint32_t AVInput::writeEDIDWrapper(const JsonObject& parameters, JsonObject& response)
+{
+    LOGINFOMETHOD();
+
+    int deviceId;
+    std::string message;
+
+    if (parameters.HasLabel("deviceId") && parameters.HasLabel("message")) {
+        deviceId = parameters["deviceId"].Number();
+        message = parameters["message"].String();
+    }
+    else {
+        LOGWARN("Required parameters are not passed");
+        return Core::ERROR_BAD_REQUEST;
+    }
+
+    writeEDID(deviceId, message);
+    return Core::ERROR_NONE;
+}
+
+uint32_t AVInput::readEDIDWrapper(const JsonObject& parameters, JsonObject& response)
+{
+    LOGINFOMETHOD();
+
+    int portId = 0;
+    try {
+        portId = parameters["portId"].Number();
+    }catch (...) {
+            LOGWARN("Invalid Arguments");
+            return Core::ERROR_BAD_REQUEST;
+    }
+
+    string edid = readEDID (portId);
+    if (edid.empty()) {
+        return Core::ERROR_GENERAL;
+    }
+    else {
+        response["EDID"] = edid;
+        return Core::ERROR_NONE;
+    }
+}
+
+JsonArray AVInput::getInputDevices(int iType)
+{
+    JsonArray list;
+    try
+    {
+        int num = 0;
+        if (iType == HDMI) {
+            num = device::HdmiInput::getInstance().getNumberOfInputs();
+        }
+        else if (iType == COMPOSITE) {
+            num = device::CompositeInput::getInstance().getNumberOfInputs();
+        }
+        if (num > 0) {
+            int i = 0;
+            for (i = 0; i < num; i++) {
+                //Input ID is aleays 0-indexed, continuous number starting 0
+                JsonObject hash;
+                hash["id"] = i;
+                std::stringstream locator;
+                if (iType == HDMI) {
+                    locator << "hdmiin://localhost/deviceid/" << i;
+                    hash["connected"] = device::HdmiInput::getInstance().isPortConnected(i) ? "true" : "false";
+                }
+                else if (iType == COMPOSITE) {
+                    locator << "cvbsin://localhost/deviceid/" << i;
+                    hash["connected"] = device::CompositeInput::getInstance().isPortConnected(i) ? "true" : "false";
```

Comment:
please use boolean if it is boolean. Do not use string for boolean

Reply:
Modified code to assign boolean value for connected